### DISCUSSION TITLE
Sandbox app opens http url in firefox which is in another sandbox

### DISCUSSION THREAD

**Discussion by vukitoso** [netblue30/firejail#4756] [Q&A] [answered]
Hi.
firejail version 0.9.67

Telegram sandbox:
"private-bin telegram"

Firefox sandbox:
"private-bin firefox"

I launched telegram application and when opening http links they are opened in firefox browser (in the appropriate sandbox).
Why? After all, there are no other applications in the telegram sandbox. Nothing should open at all.

**Comment by kmk3** [comment, marked as answer]
@vebmaster [on Dec 9](https://github.com/netblue30/firejail/discussions/4756#discussion-3732116):

> Hi. firejail version 0.9.67

Which distro and version?

> Telegram sandbox: "private-bin telegram"
> 
> Firefox sandbox: "private-bin firefox"
> 
> I launched telegram application and when opening http links they are opened
> in firefox browser (in the appropriate sandbox). Why? After all, there are no
> other applications in the telegram sandbox. Nothing should open at all.

They probably communicate through dbus, which is not completely blocked on all
profiles, precisely to allow that to happen (for usability reasons).  Example:

If you do not want to allow dbus anywhere, add the following to e.g.:
~/.config/firejail/globals.local:

```firejail
dbus-user none
dbus-system none
```

Note: That may break things like notifications and system tray icons.

Related:

* <https://github.com/netblue30/firejail/pull/3265>
* <https://github.com/netblue30/firejail/issues/3471>
* <https://github.com/netblue30/firejail/issues/3919>
* <https://github.com/glitsj16/firejail-handler-http>
* <https://github.com/glitsj16/firejail-handler-extra>

**Reply by vukitoso** [reply]
thx.

> Which distro and version?

Firejail build from source code
OS: debian 10

**Comment by vukitoso** [comment]
@kmk3, please tell me one moment:
I found the following line in the profile config:
```
?ALLOW_TRAY: dbus-user.talk org.kde.StatusNotifierWatcher
```
In the documentation, I did not find a mention of this config / parameter format.
How to understand this entry correctly?

**Reply by rusty-snake** [reply]
> In the documentation, I did not find a mention of this config / parameter format.

If you search for `ALLOW_TRAY`, you will find

> **?CONDITIONAL: profile line**
>   Conditionally add profile line.
>
>   Example: "?HAS_APPIMAGE: whitelist ${HOME}/special/appimage/dir"
>
>   This example will load the whitelist profile line only if the --appimage option has been specified on the command line.
>
>   Currently the only conditionals supported this way are HAS_APPIMAGE, HAS_NET, HAS_NODBUS, HAS_NOSOUND, HAS_PRIVATE and HAS_X11. The conditionals ALLOW_TRAY, BROWSER_DISABLE_U2F and BROWSER_ALLOW_DRM can  be  enabled  or  disabled globally in Firejail's configuration file.
>
>   The profile line may be any profile line that you would normally use in a profile except for "quiet" and "include" lines.

**Reply by vukitoso** [reply]
thx
